### ISSUE TITLE
Improve checkbox groupings in Datagrid customize columns tearsheet

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#4796]
Reported by @kristastarr in the Datagrid [customize columns review](https://github.ibm.com/CDAI-design/pal/issues/1576) from the accessibility audit.

Flagged as a potential violation.

> **Verify that this un-named, ungrouped checkbox input is not related to other checkboxes**
> Related sets of radio buttons or checkboxes should be programmatically grouped

Select all checkbox in the Create tearsheet.

![image](https://github.com/carbon-design-system/ibm-products/assets/1717398/18feaa3c-ba74-4570-a72a-856eec843896)

We made need to group the column/row checkboxes in the tearsheet, while adding a separate name for the select all checkbox.

> - Group related controls of the same type by enclosing them within a `<fieldset>` element
> - **Or**, group related checkboxes by using `role="group"`
> - **Or**, disassociate unrelated controls or controls of different types by using different `name` attribute values

> Rule ID: input_checkboxes_grouped
> Reason ID: Potential_UnnamedCheckbox

**Comment by ljcarot** [comment]
Since we are moving to TanStack Table, closing this issue as not planned.
